genesis/example/member.go

```diff
@@ -17,134 +17,140 @@
 package example
 
 import (
+	"crypto/rand"
+	"crypto/rsa"
 	"fmt"
 
 	"github.com/insolar/insolar/genesis/model/class"
-	"github.com/insolar/insolar/genesis/model/domain"
+	"github.com/insolar/insolar/genesis/model/contract"
 	"github.com/insolar/insolar/genesis/model/factory"
 	"github.com/insolar/insolar/genesis/model/object"
 	"github.com/insolar/insolar/genesis/model/resolver"
 )
 
-// MemberDomainName is a name for member domain.
-const MemberDomainName = "MemberDomain"
+type Member interface {
+	object.ComposingContainer
+	contract.SmartContract
+	GetUsername() string
+	GetPublicKey() *rsa.PublicKey
+}
 
-// MemberDomain is a contract that allows to add new members to system.
-type MemberDomain interface {
-	// Base domain implementation.
-	domain.Domain
-	// CreateMember is used to create new member as a child to domain storage.
-	CreateMember(factory.Factory) (string, error)
-	// GetMember returns member from its record in domain storage.
-	GetMember(string) (resolver.Proxy, error)
+type member struct {
+	contract.BaseSmartContract
+	Username   string
+	privateKey *rsa.PrivateKey
```

Comment:
no private key stored in blockchain, it is only passed when node is going to be initialized